src/support/thread_group.c

```diff
@@ -334,3 +335,30 @@ __wt_thread_group_start_one(
 
 	return (ret);
 }
+
+/*
+ * __wt_thread_group_stop_one --
+ *	Stop one thread if possible.
+ */
+int
+__wt_thread_group_stop_one(
+    WT_SESSION_IMPL *session, WT_THREAD_GROUP *group, bool wait)
+{
+	WT_DECL_RET;
+
+	if (group->current_threads <= group->min)
+		return (0);
+
+	if (wait)
```

Comment:
I spent some time looking at places we create and destroy threads.
Apart from the usual create/destroy the thread group and reconfigure, it looks like __evict_pass() is the only caller that could have modified the number of threads.
Since with this change __evict_pass() can only create/delete threads in the context of thread[0], do we have to worry about thread group being modified with parallel threads. If not we can do away with the group->lock.